### PR TITLE
Added new implementation of serialize_ICON

### PR DESCRIPTION
**Pull request by Oddan**
The new function has been added to the file WriteRestartHelpers, and intended to take over for the local function 'serialize_ICON' in `restartIO.cpp` when `restartIO::save()` is to be updated.  The purpose of the new implementation is to be compatible with Eclipse.

This pull request builds on previous pull request 392 by Joakim Hove.

### CODE REVIEW

File: src/opm/output/eclipse/WellDataSerializers.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+#include <ert/ecl_well/well_const.h> // containts ICON_XXX_INDEX
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/output/eclipse/WriteRestartHelpers.hpp>
+#include <vector>
+
+// ----------------------------------------------------------------------------
+std::vector<int>
+Opm::RestartIO::Helpers::
+serialize_ICON(int report_step,
```
Comment (joakim-hove) [inline src/opm/output/eclipse/WellDataSerializers.cpp]: See the discussion in #392 of the different time coordinates; I am quite certain this should be what is named `lookup_step` in that PR.

File: src/opm/output/eclipse/WellDataSerializers.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            const size_t offset = well_offset + completion_offset;
+
+            data[ offset + ICON_IC_INDEX ] = completion.complnum();
+            data[ offset + ICON_I_INDEX ] = completion.getI() + 1;
+            data[ offset + ICON_J_INDEX ] = completion.getJ() + 1;
+            data[ offset + ICON_K_INDEX ] = completion.getK() + 1;
+            data[ offset + ICON_DIRECTION_INDEX ] = completion.getDirection();
+            data[ offset + ICON_STATUS_INDEX ] =
+                (completion.getState() == WellCompletion::StateEnum::OPEN) ?
+                1 : -1000;
```
Comment (joakim-hove) [inline src/opm/output/eclipse/WellDataSerializers.cpp]: Ehhh - I think closed is just 0?

**Comment by joakim-hove** [comment]
OK; this is fine  - when the `report_step` has been changed to `lookup_step` it can be merged. I will leave that merging to @atgeirr when #392 - or a variety of that is ready.

File: src/opm/output/eclipse/WellDataSerializers.cpp (hunk 2 above)
Comment (Oddan) [reply src/opm/output/eclipse/WellDataSerializers.cpp]: I agree, but I chose to stick to the python script, which for some reason used this constant.

**Comment by atgeirr** [comment]
Superseded by #399 which contains this, closing.

**Pull request closed without merging**